src/components/expression/expression.js

```diff
@@ -172,6 +204,22 @@ genotet.ExpressionView = function(viewName, params) {
     .on('genotet.updatePanel', function() {
       this.panel.dataLoaded();
     }.bind(this));
+
+  // Set up link callbacks.
+  $(this).on('genotet.addProfile', function(event, data) {
+    var genes = data.id.split(',');
```

Comment:
You seem to be misunderstanding my point. This is not related to data retrieval at all. I'm just saying if you can send data.id which is a comma separated string, why not sending an array of strings?